### DISCUSSION TITLE
convert list of records to table

### DISCUSSION THREAD

**Discussion by N4tus** [nushell/nushell#6068] [Q&A]
I have a list of records (`let pp3 = (ls | get name | where ($it | str ends-with '.pp3') | each {|it| {img: ($it | parse '{img}.pp3' | get 0.img), pp3: $it}})`) and I would like to work with tables. The only way I found was use reduce and merge all elements into a table with two columns 'img' and 'pp3'. 

The 'pp3' will always stay the same as the name from the ls command,

**Comment by fdncred** [comment]
I think if you wrap your record in brackets, it'll turn it into a table. So just put [ and ] around your { and }.

**Reply by N4tus** [reply]
`[(ls | get name | where ($it | str ends-with '.pp3') | each {|it| {img: ($it | parse '{img}.pp3' | get 0.img), pp3: $it}})] | get 0` produces a table, but I feel like there's a batter way

**Comment by fdncred** [comment]
Now that I'm not on my phone, I'm not following because this already produces a table.
```
ls | get name | where ($it | str ends-with '.rs') | each {|it| {img: ($it | parse '{img}.rs' | get 0.img), pp3: $it}}
```
You can see it's a table by putting a `| describe` on the end.

But in my first comment I was suggesting that you wrap your record creation in brackets.
```
ls | get name | where ($it | str ends-with '.rs') | each {|it| [{img: ($it | parse '{img}.rs' | get 0.img), pp3: $it}]}
```
